host/crypto/openssl/key.c

```diff
@@ -399,6 +480,125 @@ oe_result_t oe_private_key_sign(
     return result;
 }
 
+oe_result_t oe_sign_by_engine(
+    const char *engine_id,    // string name of the engine, ie "esrp"
+    const char *key_id,       // string id of the desired signing key, ie "KEYCODE=CP-23072"
+    const char *engine_load_path, // Location of the 
+    oe_hash_type_t hash_type,
+    const void* hash_data,
+    size_t hash_size,
+    uint8_t* signature,
+    size_t* signature_size)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    EVP_PKEY_CTX* ctx = NULL;
+    EVP_PKEY *pkey    = NULL;
+    ENGINE *e = NULL;
+
+    /* Check for null parameters */
+    if (!engine_id || !key_id  || !hash_data || !hash_size ||
+        !signature_size)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    /* Check that hash buffer is big enough (hash_type is size of that hash) */
+    if (hash_type > hash_size)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    /* If signature buffer is null, then signature size must be zero */
+    if (!signature && *signature_size != 0)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    /* Initialize OpenSSL */
+    oe_initialize_openssl();
+
+    /* Load and init the engine. */
+    if (engine_load_path)
+    {
+        /* If the engine load path is set this is a dynamically loaded engine, not built in */
+        ENGINE_load_dynamic();
+        ENGINE *dyne = ENGINE_by_id("dynamic");
+        
+        if (!ENGINE_ctrl_cmd_string(dyne, "ID", "esrp", 0))
+            OE_RAISE(OE_CRYPTO_ERROR);
+
+        if (!ENGINE_ctrl_cmd_string(dyne, "SO_PATH", engine_load_path, 0))
+            OE_RAISE(OE_CRYPTO_ERROR);
+
+        if (!ENGINE_ctrl_cmd_string(dyne, "LIST_ADD", "1", 0))
+            OE_RAISE(OE_CRYPTO_ERROR);
+
+        if (!ENGINE_ctrl_cmd_string(dyne, "LOAD", NULL, 0))
+            OE_RAISE(OE_CRYPTO_ERROR);
+    }
+
+    if (!( e = ENGINE_by_id(engine_id)))
+        OE_RAISE(OE_CRYPTO_ERROR);
+
+    if (!ENGINE_init(e))
+    {
+        /* if ENGINE_init failed there is no valid reference to e */
+        e = NULL;  
+        OE_RAISE(OE_CRYPTO_ERROR);
+    }
+
+    /* Then get the private key 2do: try to sign using the engine rather 
```

Comment:
what does "2do" mean?   Is this a "TODO"?   Be explicit to make todos easier to find.
Even better would be to file a github issue and mention the # in the comment.

Reply:
I rephrased, though it seems a reasonable thing to say.
What do I use to indicate that something could be better but the surrounding environment does not yet enable it, so its not currently practical?